metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java

```diff
@@ -306,15 +307,17 @@ public Index parseIndex(T object) {
                 }
 
                 if (null != fields && !fields.isEmpty()) {
-                    List<SortKey> f = new ArrayList<>();
+                    List<IndexSortKey> f = new ArrayList<>();
 
                     for (T s : fields) {
                         String fld = getRequiredStringProperty(s, "field");
                         String dir = getStringProperty(s, "dir");
+                        String ci = getStringProperty(s, "caseInsensitive");
```

Comment:
It is optional though, so cast it to Boolean and check null.

On Mon, Feb 15, 2016 at 11:49 AM, Brandon Vulaj <notifications@github.com>
wrote:

> In
> metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java
> <https://github.com/lightblue-platform/lightblue-core/pull/582#discussion_r52932421>
> :
>
> >
> >                      for (T s : fields) {
> >                          String fld = getRequiredStringProperty(s, "field");
> >                          String dir = getStringProperty(s, "dir");
> > +                        String ci = getStringProperty(s, "caseInsensitive");
>
> Nope, the typecast exception is thrown right at this line.
>
> Changing that to the following is enough: String cannot be case to
> boolean, blah, blah.
>
> boolean ci = (boolean) getValueProperty(s, "caseInsensitive");
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/582/files#r52932421>
> .
>